### ISSUE TITLE
[JENKINS-24904] Add option to archive artifacts in the background

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#23477]
<p>I have a job that takes about 2 hours to run — 15 minutes to build, 1 hour to run automated tests, and 45 minutes to archive the artifacts (the node's upstream internet connection isn't very fast).</p>

<p>Would it make sense to add the option to start archiving the artifacts right after a certain build step completes, and archive in parallel with the remaining build steps?  In a case like this, it could reduce the job from 2 hours to 1 hour 15 minutes.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/smokris.png" title="smokris's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=smokris">smokris</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-24904" target="_blank">Add option to archive artifacts in the background</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>label(s)</b>: performance
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2017-03-09T04:43:58+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I have a job that takes about 2 hours to run — 15 minutes to build, 1 hour to run automated tests, and 45 minutes to archive the artifacts (the node's upstream internet connection isn't very fast).</p>

<p>Would it make sense to add the option to start archiving the artifacts right after a certain build step completes, and archive in parallel with the remaining build steps?  In a case like this, it could reduce the job from 2 hours to 1 hour 15 minutes.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-24904?focusedId=216728&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-216728">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I don't think this would make sense. Too error prone.</p>
</pre>
</details>

</ul>
</details>

<p>I don't think this would make sense. Too error prone.</p>
